### PR TITLE
Switch over to Github actions

### PR DESCRIPTION
**Pull request by shigmas**
Running inside the github containers seemed to lock up the test MQTT server.
 - Made a mutex for each variable that is being guarded and made them RW
 - Occasional errors in one of the test, so I removed the check. It was for the test server, so it's okay to ignore.

v2 tests are still locking up, but we won't be doing further development, so I don't think it's worth keeping the CircleCI config just to run those. I'll put a little more time into tracking down the problem and create another PR for those tests.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Set up Go 1.13.3
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13.3
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
```
Comment (xorphitus) [inline .github/workflows/ci.yml:28]: I recommend you to cache dependencies, like
https://github.com/mvdan/github-actions-golang#how-do-i-set-up-caching-between-builds

It uses `go.sum`'s hash (maybe md5) for a cache key. It enables us to drop the cache when dependencies are changed.

Comment (xorphitus) [reply .github/workflows/ci.yml:28]: The cache space is limited to 5GB and the following is GitHub Actions eviction policy.
https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

**Review by xorphitus — COMMENTED** [review]
Nice! I made just one suggestion.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shigmas) [reply .github/workflows/ci.yml:28]: I see. thanks!

**Review by xorphitus — APPROVED** [review]

**Pull request merged**